torch/distributions/transforms.py

```diff
@@ -60,8 +60,13 @@ class Transform(object):
             the codomain. Transforms that are not bijective should at least
             maintain the weaker pseudoinverse properties
             ``t(t.inv(t(x)) == t(x)`` and ``t.inv(t(t.inv(y))) == t.inv(y)``.
+        event_dim (int): Number of dimensions in the transform ``event_shape``.
```

Comment:
Sorry, can you please quickly remind me what is a transform's `event_shape`? It gives some examples, but I have no idea where does it come from

Reply:
The `event_shape` is the shape of the smallest slice of a random variable that jointly distributed. For univariate distributions, `event_shape == ()`. For multivariate distributions over vectors like `MultivariateNormal`, `event_shape == (n,)` for some n. For distributions over square matrices like `InverseWishart`, `event_shape == (n,n)`.

Each distribution has a total shape = `batch_shape + event_shape`: the batch part is over independent random variables, and the event part is over jointly distributed random variables.

This matters because `.log_prob()` should return something of `batch_shape`: each chunk of `event_shape` defines a single number in `.log_prob()`. That is the crux of this PR :smile:

Reply:
But how is that connected with transforms? Do `event_shapes` have to match, or can you add more dependent axes because the transforms operates e.g. on vectors?

Reply:
`event_shape`s do not need to match, but `TransformedDistribution.log_prob()` does need to know the largest `event_dim` among its `.base_dist` and its `.transforms`. This largest `event_dim` determines the `TransformedDistribution.event_shape`. For example:
- you can apply an `AffineTransform` with `event_dim == 0` to a `MultivariateNormal` with `len(event_shape) == 1`. The resulting `TransformedDistribution` has `len(event_shape) == 1` and needs to know to reduce the rightmost dim of `.transforms[0].log_det_abs_jacobian(...)` during `TransformedDistribution.log_prob()`
- you can apply a `BoltzmanTransform` with `event_dim == 1` to a `Normal` with `len(event_shape) == 0`. The resulting `TransformedDistribution` also has `len(event_shape) == 1` and needs to reduce the rightmost dim of `base_dist.log_prob()` during `TransformedDistribution.log_prob()`.

Reply:
Ok, that makes sense. To sum up, it's the number of dimensions that were correlated either by the base distribution, or by any of the transforms, correct? Can you please add a note like this in the docs? I think it would make it slightly clearer

Reply:
Well said. I've updated the docs of `Transform` and `TransformedDistribution`.

Reply:
Thank you!